File: library/src/main/java/com/github/sundeepk/compactcalendarview/domain/Event.java
package com.github.sundeepk.compactcalendarview.domain;

import android.graphics.Bitmap;
import android.support.annotation.Nullable;

public class Event {

    private int color;
    private long timeInMillis;
    private Object data;
    private Bitmap icon;

    public Event(int color, long timeInMillis) {
        this.color = color;
        this.timeInMillis = timeInMillis;
    }

    public Event(int color, long timeInMillis, Object data, Bitmap icon) {
        this.color = color;
        this.timeInMillis = timeInMillis;
        this.data = data;
        this.icon = icon;
    }

    public int getColor() {
        return color;
    }

    public long getTimeInMillis() {
        return timeInMillis;
    }

    @Nullable
    public Bitmap getIcon() {
        return icon;
    }

    @Nullable
    public Object getData() {
        return data;
    }

    @Override
    public String toString() {
        return "Event{" +
                "color=" + color +
                ", timeInMillis=" + timeInMillis +
                ", data=" + data +
                ", icon=" + icon +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Event event = (Event) o;

        if (color != event.color) return false;
        if (timeInMillis != event.timeInMillis) return false;
        if (data != null ? !data.equals(event.data) : event.data != null) return false;
        return icon != null ? icon.equals(event.icon) : event.icon == null;

    }

    @Override
    public int hashCode() {
        int result = color;
        result = 31 * result + (int) (timeInMillis ^ (timeInMillis >>> 32));
        result = 31 * result + (data != null ? data.hashCode() : 0);
        result = 31 * result + (icon != null ? icon.hashCode() : 0);
        return result;
    }

}

File: library/src/main/java/com/github/sundeepk/compactcalendarview/domain/Event.kt
package com.github.sundeepk.compactcalendarview.domain

class Event {
    var color: Int
        private set
    var timeInMillis: Long
        private set
    var data: Any? = null
        private set

    constructor(color: Int, timeInMillis: Long) {
        this.color = color
        this.timeInMillis = timeInMillis
    }

    constructor(color: Int, timeInMillis: Long, data: Any?) {
        this.color = color
        this.timeInMillis = timeInMillis
        this.data = data
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val event = o as Event
        if (color != event.color) return false
        if (timeInMillis != event.timeInMillis) return false
        return !if (data != null) data != event.data else event.data != null
    }

    override fun hashCode(): Int {
        var result = color
        result = 31 * result + (timeInMillis xor (timeInMillis ushr 32)).toInt()
        result = 31 * result + if (data != null) data.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "Event{" +
                "color=" + color +
                ", timeInMillis=" + timeInMillis +
                ", data=" + data +
                '}'
    }
}
